### PR TITLE
etcdmain: add 'version' sub-command

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by gyuho** [comment]
```
./bin/etcd version    # or ./bin/etcd --version
etcd version: 3.0.0+git
Git SHA:      581f847
Go version:   devel +7ea6212 Thu Jun 30 21:09:56 2016 +0000
Go OS/Arch:   linux/amd64
Built:        Thu Jun 30 21:59:32 PDT 2016
```

/cc @heyitsanthony @xiang90 

Also I am trying to see if we can auto-generate help page for our etcd sub commands. Too much work to maintain it manually.

File: etcdmain/version.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"fmt"
+	"os"
+	"runtime"
+
+	"github.com/coreos/etcd/version"
+	"github.com/spf13/cobra"
+)
+
+func printVersion() {
```
Comment (xiang90) [inline etcdmain/version.go]: probably move this to pkg version/version.go

File: etcdmain/config.go [hunk 2]
```diff
@@ -279,15 +277,14 @@ func (cfg *config) Parse(arguments []string) error {
 		os.Exit(2)
 	}
 	if len(cfg.FlagSet.Args()) != 0 {
+		if cfg.FlagSet.Arg(0) == "version" {
```
Comment (xiang90) [inline etcdmain/config.go]: this is a wrong place to do the sub command thing (this needs to be done in main.go as other sub commands). also i think probably etcd --version is good enough for now.

**Comment by gyuho** [comment]
ok got it. maybe we can add later if needed. thanks.

**Pull request closed without merging**